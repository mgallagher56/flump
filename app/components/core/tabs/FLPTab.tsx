import React from 'react';

import type { TabProps } from '@chakra-ui/react';
import { Button, useMultiStyleConfig } from '@chakra-ui/react';

import FLPBox from '../structure/FLPBox';

const FLPTab: React.FC<TabProps> = (props) => {
  const isSelected = !!props['aria-selected'];

  const styles = useMultiStyleConfig('Tabs', { props });

  return (
    <Button __css={styles.tab} {...props} width="max-content">
      <FLPBox as="span" mr="2">
        {isSelected ? '😎' : '😐'}
      </FLPBox>
      {props.children}
    </Button>
  );
};
export default FLPTab;
