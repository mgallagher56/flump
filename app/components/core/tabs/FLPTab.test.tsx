import React from 'react';

import { Tabs } from '@chakra-ui/react';
import { render } from '@testing-library/react';
import { describe, expect, test } from 'vitest';

import FLPTab from './FLPTab';

const tabWrapper = (children: React.ReactNode): React.ReactElement => <Tabs>{children}</Tabs>;

describe('FLPTab', () => {
  test('renders correctly', () => {
    const { container } = render(
      tabWrapper(
        <>
          <FLPTab aria-selected>Tab 1</FLPTab>
          <FLPTab>Tab 2</FLPTab>
          <FLPTab>Tab 3</FLPTab>
        </>
      )
    );
    expect(container).toMatchSnapshot();
  });
});
