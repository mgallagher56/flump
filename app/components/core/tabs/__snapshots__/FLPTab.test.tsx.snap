// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FLPTab > renders correctly 1`] = `
<div>
  <div
    class="chakra-tabs css-0"
  >
    <button
      aria-selected="true"
      class="chakra-button css-cf3le8"
      type="button"
    >
      <span
        class="css-1qkvt8e"
      >
        😎
      </span>
      Tab 1
    </button>
    <button
      class="chakra-button css-cf3le8"
      type="button"
    >
      <span
        class="css-1qkvt8e"
      >
        😐
      </span>
      Tab 2
    </button>
    <button
      class="chakra-button css-cf3le8"
      type="button"
    >
      <span
        class="css-1qkvt8e"
      >
        😐
      </span>
      Tab 3
    </button>
  </div>
</div>
`;
